x-pack/plugins/fleet/server/saved_objects/index.ts

```diff
@@ -143,6 +143,20 @@ const getSavedObjectTypes = (
         is_preconfigured: { type: 'boolean', index: false },
         ssl: { type: 'binary' },
         proxy_id: { type: 'keyword' },
+        shipper: {
+          properties: {
+            disk_queue_enabled: { type: 'boolean', index: false },
+            disk_queue_path: { type: 'keyword', index: false },
+            disk_queue_max_size: { type: 'integer', index: false },
```

Comment:
@rudolf all nested props under `shipper` are meant to not be indexed. Are we fine with manual definition for each field, or does this still consumes more resources or mapping size compared to having the root `shipper` field non-indexed?

Reply:
yes, the only good reason to specify `index: false` fields is if it's impossible to do any kind of validation in other layers and you might be worried that a user would inject data into your saved objects that could break your application.

It looks like there's an API layer with validation in front of these saved objects. They are still exposed through the generic saved objects APIs so it's possible (though unlikely) that a user could circumvent the API validation. Today it's possible to set `hidden: true` on the saved objects but it requires some additional changes. We'll be making that easier very soon https://github.com/elastic/kibana/issues/147150.

So I would recommend that we remove these fields and rather specify `dynamic: false`. And then as soon as https://github.com/elastic/kibana/issues/147150 is complete disable the saved objects HTTP APIs for these types.

Reply:
So, if I understand correctly, these props would look like this?

```
   shipper: {
          dynamic: false,
          properties: {
            disk_queue_enabled: { type: 'boolean' },
            disk_queue_path: { type: 'keyword' },
            disk_queue_max_size: { type: 'integer' },
            disk_queue_encryption_enabled: { type: 'boolean' },
            disk_queue_compression_enabled: { type: 'boolean' },
            compression_level: { type: 'integer' },
            loadbalance: { type: 'boolean' },
            mem_queue_events: { type: 'integer' },
            queue_flush_timeout: { type: 'integer' },
            max_batch_bytes: { type: 'integer' },
          },
        },
```

Reply:
I pushed a change with `dynamic: false` on the shipper property. I'll merge this PR as it is, there will likely be a follow up before FF so if any change is necessary I'll address it there.